### PR TITLE
penguins: add coordinate transformation formulas

### PR DESCRIPTION
**Pull request by jnccd**
Adds additional info to help students

### CODE REVIEW

File: hyperbook/book/spiele/penguins/xml.md [hunk 1]
```diff
@@ -131,8 +131,18 @@ Der Inhalt der `field` Objekte steht für die Anzahl Fische auf dem Feld, oder f
...
 
+##### Koordinatensysteme
+
 Beachtet, dass der XML-Status [odd-r Koordinaten](https://www.redblobgames.com/grids/hexagons/#neighbors-offset) nutzt, und dass die Koordinaten in Zügen als [doubled Koordinaten](https://www.redblobgames.com/grids/hexagons/#neighbors-doubled) interpretiert werden.
 
+Die Formeln für die Transformation von dem einen System in das andere sind folgendermaßen:
+
+odd-r zu doubled: ``(x, y) => (x * 2 + y % 2, y)``
+
+doubled zu odd-r: ``(x, y) => (x / 2 - y % 2, y)``
```
Comment (FalconsSky) [inline hyperbook/book/spiele/penguins/xml.md:142]: It should perhaps be noted here that the result must be rounded up?

File: hyperbook/book/spiele/penguins/xml.md [hunk 2]
```diff
@@ -131,8 +131,18 @@ Der Inhalt der `field` Objekte steht für die Anzahl Fische auf dem Feld, oder f
...
+
+Beachtet weiter, dass im doubled Koordinatensystem unabhängig der Größe (0, 1), (0, 3), (0, 5), etc. immer out of bounds sind, da sie in odd-r zu (-1, y) übersetzt werden.
```
Comment (FalconsSky) [inline hyperbook/book/spiele/penguins/xml.md:144]: > Beachtet weiter, dass im doubled Koordinatensystem unabhängig der Größe, die Koordinaten (0, 1), (0, 3), (0, 5) et cetera, immer unzulässig sind, da sie in odd-r zu (-1, y) übersetzt werden.

Maybe you should wrap this information in a [alert](https://hyperbook.openpatch.org/elements/alert):
```
:::alert{warn}
Warn
:::
```

**Review by FalconsSky — COMMENTED** [review]
This two things should be worked on IMO.

File: hyperbook/book/spiele/penguins/xml.md (hunk 2 above)
Comment (jnccd) [reply hyperbook/book/spiele/penguins/xml.md:145]: Done

File: hyperbook/book/spiele/penguins/xml.md (hunk 1 above)
Comment (jnccd) [reply hyperbook/book/spiele/penguins/xml.md:142]: up? Not down?
In the example here it looks like the first row should be rounded to X=0
image
Anyway most programming languages automatically round down for integers. Nonetheless adding a comment could be useful.

Comment (jnccd) [reply hyperbook/book/spiele/penguins/xml.md:142]: Btw ist hier umgesetzt 
https://github.com/software-challenge/docs/pull/66/commits/0a71d4edae2bbb332fc6d8d02eb6284d3911b198

Comment (FalconsSky) [reply hyperbook/book/spiele/penguins/xml.md:142]: Well the problem that occurs with that is, whe you have for instance the coordinate `(3, 3)`.
The calculation would be: `(3, 3)->(3 / 2 - 3 % 2, 3)->(0.5, 3)` and that would be `(0, 3)` when cast into integer, what is not correct.

Comment (jnccd) [reply hyperbook/book/spiele/penguins/xml.md:142]: Ahh right, I didnt think about the -1
I guess Ill have to fix some code later then

**Review by FalconsSky — APPROVED** [review]
Looks good to me.

**Pull request merged**